src/DSharp.Compiler/Generator/ExpressionGenerator.cs

```diff
@@ -615,6 +630,13 @@ public static void GenerateExpression(ScriptGenerator generator, MemberSymbol sy
                 GenerateExpressionList(generator, symbol, expression.Indices);
                 writer.Write(")");
             }
+            else if (expression.Indexer.Parent is TypeSymbol typeSymbol && typeSymbol.IsNativeObject())
```

Comment:
what case does this cover?

Reply:
This covers where we have a type like the following:
```
[ScriptImport]
    [ScriptIgnoreNamespace]
    [ScriptName("Object")]
    public class JsObject
    {
        public extern object this[string propertyName]
        {
            get; set;
        }
    }
```
And ideally when we access the indexer I would expect it to do this `obj[someindex]`
whereas currently it does this: `obj.get_item(someIndex);`
This causes all of our cloning to explode.